include/translatable-object.php

```diff
@@ -460,30 +466,41 @@ public function sanitize_int_ids_list( $ids ) {
 	}
 
 	/**
-	 * Returns SQL query that fetches the IDs of the objects without language.
+	 * Fetches the IDs of the objects without language.
 	 *
 	 * @since 3.4
+	 * @since 3.7 Returns the result of the query instead of the SQL query only.
 	 *
 	 * @param int[] $language_ids List of language `term_taxonomy_id`.
 	 * @param int   $limit        Max number of objects to return. `-1` to return all of them.
 	 * @param array $args         The object args.
-	 * @return string
+	 * @return string[]
 	 *
 	 * @phpstan-param array<positive-int> $language_ids
 	 * @phpstan-param -1|positive-int $limit
 	 * @phpstan-param array<empty> $args
 	 */
 	protected function get_objects_with_no_lang_sql( array $language_ids, $limit, array $args = array() ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		global $wpdb;
+
 		$db = $this->get_db_infos();
 
-		return sprintf(
-			"SELECT {$db['table']}.{$db['id_column']} FROM {$db['table']}
-			WHERE {$db['table']}.{$db['id_column']} NOT IN (
-				SELECT object_id FROM {$GLOBALS['wpdb']->term_relationships} WHERE term_taxonomy_id IN (%s)
+		return $wpdb->get_col(
+			$wpdb->prepare(
+				sprintf(
+					"SELECT %%i FROM %%i
+					WHERE %%i NOT IN (
+						SELECT object_id FROM {$wpdb->term_relationships} WHERE term_taxonomy_id IN (%s)
+					)
+					LIMIT %%d",
+					implode( ',', array_fill( 0, count( $language_ids ), '%d' ) )
+				),
+				array_merge(
+					array( $db['id_column'], $db['table'], $db['id_column'] ),
+					$language_ids,
+					array( $limit >= 1 ? $limit : 4294967295 )
```

Comment:
`$limit` could erase a value in `$language_ids` here, right?
Would it be better to concatenate (`+`) arrays instead of merging them?

Reply:
We need to ensure that we have keep the expected number of elements. `+` doesn't ensure that unlike `array_merge()` (because I used numeric keys).